screens/UserScreens/ProductionDashboard.jsx

```diff
@@ -0,0 +1,111 @@
+import { useState, useEffect } from "react";
+import { View, Text, ActivityIndicator } from "react-native";
+import { get, ref, onValue } from "firebase/database";
+import { firebase_db } from "../../firebase.config.js";
+
+export default function ProductionDashboardScreen({ route }) {
+    const [production, setProduction] = useState({});
+    const [admins, setAdmins] = useState([]);
+    const [participants, setParticipants] = useState([]);
+    const [loading, setLoading] = useState(true);
+    const playCode = route.params.playCode;
+    const db = firebase_db;
+
+    useEffect(
+        () =>
+            onValue(
+                ref(db, `/productions/${playCode}`),
+                (snapshot) => {
+                    if (!snapshot.exists()) {
+                        setLoading(false);
+                        return;
+                    }
+                    const data = snapshot.val();
+                    setProduction(data);
+                    setLoading(false);
```

Comment:
setLoading is changed before all data collection is done

Reply:
I've stuck it in the participants onvalue though because presumably that will take longer

Reply:
Should in that case be able to stick it in a promise.all, then do a .then or something?